webui/react/src/pages/Settings/GroupManagement.tsx

```diff
@@ -0,0 +1,266 @@
+    try {
+      const response = await getGroups(
+        {
+          limit: settings.tableLimit,
+          offset: settings.tableOffset,
+        },
+        { signal: canceler.signal },
+      );
+
+      setTotal(response.pagination?.total ?? 0);
+      setGroups(
+        (response.groups || []).map((group) => ({ group })),
+      );
+    } catch (e) {
+      handleError(e, { publicSubject: 'Unable to fetch groups.' });
+    } finally {
+
+      setIsLoading(false);
+    }
+  }, [ canceler.signal,
+    settings.tableLimit,
+    settings.tableOffset,
+  ]);
+
+  const fetchUsers = useCallback(async (): Promise<void> => {
+    try {
+      const response = await getUsers(
+        {},
+        { signal: canceler.signal },
+      );
+      setUsers((prev) => {
+        if (isEqual(prev, response.users)) return prev;
+        return response.users;
+      });
+    } catch (e) {
+      handleError(e, { publicSubject: 'Unable to fetch users.' });
+    }
+  }, [ canceler.signal ]);
+
+  useEffect(() => {
+    fetchGroups();
+    fetchUsers();
+  }, [
+    settings.tableLimit,
+    settings.tableOffset,
+    fetchGroups,
+    fetchUsers ]);
+
+  const {
+    modalOpen: openCreateGroupModal,
+    contextHolder: modalCreateGroupContextHolder,
+  } = useModalCreateGroup({ onClose: fetchGroups, users: users });
+
+  const onClickCreateGroup = useCallback(() => {
+    openCreateGroupModal();
+  }, [ openCreateGroupModal ]);
+
+  const onExpand = useCallback(async (expand:boolean, record:V1GroupSearchResult) => {
+    const { group: { groupId } } = record;
+    const g = groups.find((gr) => gr.group.group.groupId === groupId);
+    if (!groupId || !expand || !g) return;
+    const response = await getGroup({ groupId });
+    g.users = response.group.users || [];
+    setGroups(JSON.parse(JSON.stringify(groups)));
+  }, [ groups ]);
+
+  const onRemoveUser = useCallback(async (record: V1GroupSearchResult, userId) => {
+    const { group: { groupId } } = record;
+    if (!groupId) return;
+    try {
+      await updateGroup({ groupId, removeUsers: [ userId ] });
+      message.success('User has been deleted.');
+      onExpand(true, record);
+      fetchGroups();
+    } catch (e) {
+      message.error('error deleting group');
+      handleError(e, { silent: true, type: ErrorType.Input });
+    }
+  }, [ onExpand, fetchGroups ]);
+
+  const expandedUserRender = useCallback((record:V1GroupSearchResult) => {
+    const { group: { groupId } } = record;
+    const g = groups.find((gr) => gr.group.group.groupId === groupId);
+    const userColumn = [ {
+      dataIndex: 'displayName',
+      key: 'displayName',
+      title: 'Display Name',
+      width: '40%',
+    }, {
+      dataIndex: 'username',
+      key: 'username',
+      title: 'User Name',
+      width: '50%',
+    }, {
+      key: 'action',
+      render: (_:string, r: V1User) => (
+        <Button onClick={() => onRemoveUser(record, r.id)}>Remove</Button>),
```

Comment:
bug: removing a record also clears the entire expanded table

Reply:
Can't believe I missed this... fixed~